src/realm/replication.cpp

```diff
@@ -62,36 +64,323 @@ Replication::version_type Replication::prepare_commit(version_type orig_version)
     return new_version;
 }
 
-void Replication::add_class(TableKey table_key, StringData, Table::Type)
+void Replication::add_class(TableKey table_key, StringData name, Table::Type type)
 {
+    if (auto logger = get_logger()) {
+        if (type == Table::Type::Embedded) {
+            logger->log(util::Logger::Level::debug, "Add %1 class '%2'", type, name);
+        }
+        else {
+            logger->log(util::Logger::Level::debug, "Add class '%1'", name);
+        }
+    }
     unselect_all();
     m_encoder.insert_group_level_table(table_key); // Throws
 }
 
-void Replication::add_class_with_primary_key(TableKey tk, StringData, DataType, StringData, bool,
+void Replication::add_class_with_primary_key(TableKey tk, StringData name, DataType pk_type, StringData pk_name, bool,
                                              Table::Type table_type)
 {
+    if (auto logger = get_logger()) {
+        logger->log(util::Logger::Level::debug, "Add %4 class '%1' with primary key property '%2' of %3",
+                    Group::table_name_to_class_name(name), pk_name, pk_type, table_type);
+    }
     REALM_ASSERT(table_type != Table::Type::Embedded);
     unselect_all();
     m_encoder.insert_group_level_table(tk); // Throws
 }
 
+void Replication::erase_class(TableKey tk, StringData table_name, size_t)
+{
+    if (auto logger = get_logger()) {
+        logger->log(util::Logger::Level::debug, "Remove class '%1'", Group::table_name_to_class_name(table_name));
+    }
+    unselect_all();
+    m_encoder.erase_class(tk); // Throws
+}
+
+
+void Replication::insert_column(const Table* t, ColKey col_key, DataType type, StringData col_name,
+                                Table* target_table)
+{
+    if (auto logger = get_logger()) {
+        const char* collection_type = "";
```

Comment:
shouldn't this be " "?

Reply:
We only want the space if the collection type string is used.